src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -211,26 +213,53 @@ public boolean addAll(Collection<? extends Student> collection) {
 
     @Override
     public boolean containsAll(Collection<?> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        for (Object e : collection)
+            if (!contains(e)) {
+                return false;
+            }
+        return true;
     }
 
     @Override
     public boolean addAll(int i, Collection<? extends Student> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        int counter = 0;
+        Iterator<? extends Student> it = collection.iterator();
```

Comment:
Can be refactored, here is pseudo code example:
1: convert collection to array
2: shift existing array by collection.size() indexes using system.arraycopy
3: write collection using system.arraycopy to allocated previously place